sets/sets.go

```diff
@@ -0,0 +1,80 @@
+// Package sets is a minimal implementation of a generic set data structure.
+
+package sets
+
+import (
+	"fmt"
+	"sort"
+
+	"golang.org/x/exp/constraints"
+)
+
+// Set is a minimal set that takes only ordered types: any type that supports the
+// operators < <= >= >.
+type Set[T constraints.Ordered] struct {
+	items map[T]struct{}
+}
+
+// New returns an empty set with capacity size. The capacity will grow and shrink as a
+// stdlib map.
+func New[T constraints.Ordered](size int) *Set[T] {
+	return &Set[T]{items: make(map[T]struct{}, size)}
+}
+
+// From returns a set from elements.
+func From[T constraints.Ordered](elements ...T) *Set[T] {
+	s := New[T](len(elements))
+	for _, i := range elements {
+		s.items[i] = struct{}{}
```

Comment:
The underlying data structure is the workhorse Go map.
The type `struct{}{}` is special and takes 0 bytes and is the idiomatic way to implement a set in Go.